monai/inferers/utils.py

```
@@ -33,12 +33,51 @@
+    layout_last = "NHWC" if t.ndim == 4 else "NDHWC"
+
+    if num_classes is not None:
+        ok = (c == num_classes) or (num_classes == 1 and c == 1)
+        if allow_binary_two_channel and num_classes == 2:
+            ok = ok or (c == 2)
+        if not ok:
+            raise ValueError(
+                f"{name}: expected {layout} with C(dim=1)==num_classes, "
+                f"but got shape={tuple(t.shape)} (C={c}) and num_classes={num_classes}. "
+                f"If your data is {layout_last}, please apply EnsureChannelFirst/EnsureChannelFirstd upstream."
+            )
```

Ruff (0.14.6)

Avoid specifying long messages outside the exception class

(TRY003)